test/tests/p5.SoundRecorder.js

```diff
@@ -106,16 +109,18 @@ describe('p5.SoundRecorder', function () {
     inputSoundFile.connect();
     inputSoundFile.loop();
     recorder.setInput();
-    recorder.record(outputSoundFile, recordingDuration, function () {
-      expect(outputSoundFile.duration()).to.eq(recordingDuration);
+    setTimeout(() => {
+      recorder.record(outputSoundFile, recordingDuration, function () {
+        expect(outputSoundFile.duration()).to.eq(recordingDuration);
 
-      const outputChannel = outputSoundFile.buffer.getChannelData(0);
-      expect(outputChannel[0]).to.not.eq(0);
+        const outputChannel = outputSoundFile.buffer.getChannelData(0);
+        expect(outputChannel[0]).to.not.eq(0);
 
-      outputSoundFile.dispose();
-      p5.prototype.outputVolume(0);
-      done();
-    });
+        outputSoundFile.dispose();
+        p5.prototype.outputVolume(0);
+        done();
+      });
+    }, 10);
```

Comment:
same question as above, it seems there is a reason you've wrapped the `recorder.record` function in a setTimeout, I wouldn't think that'd be necessary but maybe it is!

Reply:
this situation is similar to the above one. But the Time out here is set to be 10ms as the `outputChannel` without timeout is being all zeros in the returned Float32Array (size 1024) until 287th position and non-zero values after that. The weird part is that the number 287 is being constant no matter how many times I run the code. No idea why this is happening, but the timeout fixes it